hw_05_09/hw_05_10.py

```diff
@@ -0,0 +1,389 @@
+        try:
+            with open(feeds_file_path, 'r') as file:
+                words_count = {}
+                for line in file:
+                    words = re.findall(r'\b\w+\b', line.lower())
+                    for word in words:
+                        if word.isalpha():
+                            words_count[word] = words_count.get(word, 0) + 1
+                with open(csw_file_path, 'w', newline='') as csw_file:
+                    writer = csv.writer(csw_file, delimiter='-')
+                    for word, count in sorted(words_count.items()):
+                        writer.writerow([word.ljust(10), str(count).rjust(5)])
+                return "Results were updated in csw file"
+        except FileNotFoundError:
+            return "File is not found"
+
+    @staticmethod
+    def calculate_letters(feeds_file_path, csw_file_path):
+        try:
+            with open(feeds_file_path, 'r') as file:
+                all_letters_count = 0
+                letter_count = {}
+                uppercase_occurrences = {}
+                for line in file:
+                    for char in line:
+                        if char.isalpha():
+                            all_letters_count += 1
+                            char_lower = char.lower()
+                            letter_count[char_lower] = letter_count.get(char_lower, 0) + 1
+                            if char.isupper():
+                                uppercase_occurrences[char_lower] = uppercase_occurrences.get(char_lower, 0) + 1
+                for letter, count in letter_count.items():
+                    uppercase_count = uppercase_occurrences.get(letter, 0)
+                    letter_count[letter] = (count, uppercase_count)
+                # Create csv file and add results
+                with open(csw_file_path, 'w', newline='') as csw_file:
+                    writer = csv.writer(csw_file, delimiter=',')
+                    writer.writerow(['letter'.ljust(20), 'count_all'.ljust(20), 'count_uppercase'.ljust(20),
+                                     'percentage %'.ljust(20)])
+                    for letter, counts in sorted(letter_count.items()):
+                        total_count, uppercase_count = counts
+                        percentage = round(total_count / all_letters_count * 100, 2)
+                        writer.writerow([letter.ljust(20), str(total_count).ljust(20), str(uppercase_count).ljust(20),
+                                         str(percentage).ljust(20)])
+                return "Results were updated in csw file"
+        except FileNotFoundError:
+            print("Letters File not found")
+
+    @staticmethod
+    def check_for_duplicates():
+        connection = pyodbc.connect('DRIVER={SQLite3 ODBC Driver};'
+                                    'Direct=True;'
+                                    'Database=test_10.db;'
+                                    'String Types= Unicode')
+        cursor = connection.cursor()
```

Comment:
Better create separate method for DB connection to reduce duplicating the connection code in multiple methods and makes it easier to manage and update connection details.